<launch>
  <arg name="use_recovery_behaviors" default="true"/>
  
  <arg name="decimate" default="false"/>

  <arg name="gazebo" default="true"/>
  <arg name="pips_namespace" default="/move_base/PipsDWAPlannerROS"/>
  <arg name="depth_image_raw" default="/camera/depth/image_raw"/>
  <arg name="depth_info_raw" default="/camera/depth/camera_info"/>
  <arg name="rgb_image_raw" default="/camera/rgb/image_raw"/>
  <arg name="map_name" default="$(find pips_dwa_implementation)/maps/empty_world.yaml"/>

  <arg name="depth_image_topic" value="$(arg depth_image_raw)" if="$(arg gazebo)"/>
  <arg name="depth_info_topic" value="$(arg depth_info_raw)" if="$(arg gazebo)"/>

  <arg name="depth_image" default="$(arg depth_image_topic)"/>
  <arg name="depth_info" default="$(arg depth_info_topic)"/>

  <arg name="debug" default="false"/>

  <arg name="debug_prefix" value="" unless="$(arg debug)"/>
  <arg name="debug_prefix" value="xterm -e gdb --args" if="$(arg debug)"/>


  <arg name="decimation"        default="4"/>
  <arg name="decimated_image"   default="/camera/depth_decimated/image_raw"/>
  <arg name="decimated_info"    default="/camera/depth_decimated/camera_info"/>
  
  <node name="egocircle_node" pkg="pips_dwa_implementation" type="ego_circle"/>

  <include file="$(find turtlebot_navigation)/launch/includes/velocity_smoother.launch.xml"/>

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="log" clear_params="true" launch-prefix="$(arg debug_prefix)">
      <rosparam file="$(find nav_configs)/config/common_costmap_params.yaml" command="load" ns="global_costmap"/>
      <rosparam file="$(find nav_configs)/config/common_costmap_params.yaml" command="load" ns="local_costmap"/>
      <rosparam file="$(find nav_configs)/config/local_costmap_params.yaml" command="load"/>
      <rosparam file="$(find nav_configs)/config/global_costmap_params.yaml" command="load"/>
      
      <rosparam file="$(find nav_configs)/config/move_base_params.yaml" command="load"/>

      
      <!--<rosparam file="$(find nav_configs)/config/global_planner_params.yaml" command="load"/> -->
      <rosparam file="$(find nav_configs)/config/navfn_global_planner_params.yaml" command="load"/>

      <param name="base_local_planner" value="pips_dwa_implementation/DepthPipsDWAPlannerROS"/>
      <rosparam file="$(find nav_configs)/config/pips_dwa_local_planner_common_params.yaml" command="load" ns="DepthPipsDWAPlannerROS"/>
      <rosparam file="$(find nav_configs)/config/local_planner_common_params.yaml" command="load" ns="DepthPipsDWAPlannerROS"/>
      <param name="recovery_behavior_enabled" value="false" unless="$(arg use_recovery_behaviors)"/>

      <remap from="cmd_vel" to="navigation_velocity_smoother/raw_cmd_vel"/>
      
      <param name="odom_topic"                                value="/odom" />
      <param name="DepthPipsDWAPlannerROS/depth_image_topic" value="$(arg depth_image)" />
      <param name="DepthPipsDWAPlannerROS/depth_info_topic" value="$(arg depth_info)" />
      
      <param name="DepthPipsDWAPlannerROS/depth_image_cc_wrapper/depth_image_collision_checker/model_type" value="6" />
      <param name="DepthPipsDWAPlannerROS/depth_image_cc_wrapper/depth_image_collision_checker/floor_tolerance" value=".05" />
      <param name="DepthPipsDWAPlannerROS/depth_image_cc_wrapper/depth_image_collision_checker/show_im" value="true" />

      <rosparam file="$(find nav_configs)/config/turtlebot_ec_dwa_params.yaml" command="load"/>
      

  </node>

</launch>
